ext/mysqli/tests/bug81335.phpt

```diff
@@ -0,0 +1,37 @@
+--TEST--
+Bug #81335: Packets out of order after connection timeout
+--EXTENSIONS--
+mysqli
+--SKIPIF--
+<?php
+if (getenv("SKIP_SLOW_TESTS")) die("skip slow test");
+
+if (!defined('MYSQLI_STORE_RESULT_COPY_DATA')) die('skip requires mysqlnd');
+
+require_once 'connect.inc';
+if (!$link = @my_mysqli_connect($host, $user, $passwd, $db, $port, $socket)) die("skip cannot connect");
+if (mysqli_get_server_version($link) < 80024 || str_contains(mysqli_get_server_info($link), 'MariaDB')) {
+    $link->close();
```

Comment:
Here also, you can remove the unnecessary line `$link->close();` and the one below it. There is no need to close the connection here manually.

Reply:
Applied.
Do we need to skip this test for Windows due to the differences in TCP read behavior?

Reply:
I'll take a look at it later. Why is the test failing now and wasn't failing before this PR?

Reply:
This test was added in this PR and has never passed on Windows.

Reply:
> This test was added in this PR and has never passed on Windows.

yes

quote from above comment

> When a client-timeout occurs, the MySQL server sends an error 4021 and a TCP-FIN packet, and the TCP state 

> changes to FIN_WAIT_2. At this time, the TCP state of the client is CLOSE_WAIT.
> 
> MySQL-Server side: when the server TCP state is FIN_WAIT_2, the server will send a TCP RST packet after client sending a query.
> If the client is on Linux, it can still read the error message packet sent by the server.
> If the client is on Windows, an error (10053) will occur when executing receive.
> 
> I'm not sure if this is due to the TCP implementation or something else.

Reply:
> I'll take a look at it later. Why is the test failing now and wasn't failing before this PR?

Are there any updates? @kamil-tekiela

Reply:
Sorry, I forgot about this PR. I can't debug on Windows anymore so I can't check why that is failing or how to fix it. Let's add a skip for now and maybe in the future someone can debug this. Since it only bugs out in this particular scenario I hope that the impact be minimal.

Reply:
> Sorry, I forgot about this PR. I can't debug on Windows anymore so I can't check why that is failing or how to fix it. Let's add a skip for now and maybe in the future someone can debug this. Since it only bugs out in this particular scenario I hope that the impact be minimal.

Applied